### PR TITLE
Added Authentication

### PR DESCRIPTION
**Pull request by AbhishekkGautam**
### Authentication
- Login page
- Signup page

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,14 +1,22 @@
...
+  const {
+    state: { isLoggedIn },
+  } = useAuth();
   return (
     <div className="App">
+      <ToasterWrapper />
       <Routes>
         <Route path="/" element={<Home />} />
         <Route path="/trending" element={<Trending />} />
+        {!isLoggedIn && <Route path="/signup" element={<Signup />} />}
```
Comment (poornima-alugubelly) [inline src/App.js:18]: U can wrap these inside private route instead doing conditional check on each route

File: src/components/Navbar/Navbar.js [hunk 2]
```diff
@@ -1,46 +1,71 @@
...
-            placeholder="search"
-            className="search-input"
-            required
-          />
-          <button className="search-btn">
-            <span className="material-icons">search</span>
-          </button>
-        </form>
+        {pathname === "/login" || pathname === "/signup" ? null : (
+          <form action="#" className="nav-form">
```
Comment (poornima-alugubelly) [inline src/components/Navbar/Navbar.js]: I guess action=# is not needed here?

**Review by poornima-alugubelly — APPROVED** [review]

File: src/App.js (hunk 1 above)
Comment (AbhishekkGautam) [reply src/App.js:18]: login & signup are not private routes. It's a reverse implementation of the private route. if a user is logged in, then he/she can't able to visit the login/signup page. he/she has to log out first to access the login page.

File: src/components/Navbar/Navbar.js (hunk 2 above)
Comment (AbhishekkGautam) [reply src/components/Navbar/Navbar.js]: done

**Review by pritam-kr — APPROVED** [review]
Everything look good you can merge it.

**Pull request merged**